### PR TITLE
C++20 no longer captures 'this' implicitly in lambda functions.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: source/euler/initial_state_becker_solution.h [hunk 1]
```diff
@@ -113,7 +113,7 @@ namespace ryujin
           const double factor = 2. * gamma_ / (gamma_ + 1.) //
                                 * mu_ / (density_left_ * velocity_left_ * Pr);
 
-          psi = [=](double x, double v) {
+          psi = [=,this](double x, double v) {
```
Comment (bangerth) [inline source/euler/initial_state_becker_solution.h]: Not explicitly capturing `this` is deprecated in C++20 and leads to warnings.

**Comment by tamiko** [comment]
@bangerth Would you mind to fix indentation?
```diff
-          psi = [=,this](double x, double v) {
+          psi = [=, this](double x, double v) {
```
```diff
-          find_velocity = [=,this](double x) {
+          find_velocity = [=, this](double x) {
```

**Comment by tamiko** [comment]
```
/__w/ryujin/ryujin/source/euler/initial_state_becker_solution.h:116:20: error: explicit capture of 'this' with a capture default of '=' is a C++20 extension [-Werror,-Wc++20-extensions]
          psi = [=,this](double x, double v) {
```

@bangerth In `CMakeLists.txt` would you mind to simply bump `CMAKE_CXX_STANDARD` to 20?

**Comment by bangerth** [comment]
I fixed the indentation, but the other issue falls into the "ugh" category. Is the compiler saying that `[=,this]` captures the *object*, not the *pointer* by-value? Would `[this,=]` be the right way to write this?

**Comment by tamiko** [comment]
@bangerth I think you're right, this is just the order. `[this, =]` should be the correct one.

**Comment by bangerth** [comment]
OK, so done.

**Comment by tamiko** [comment]
@bangerth That didn't fix it. We could work around the issue by explicitly capturing what is used in these two functions, or by disabling the deprecation warning.

But honestly, let's just switch to C++20. *yolo*

I have pushed a commit to your branch that does just that.

**Pull request merged**